Automattic/woocommerce-payments
# Add hyphen to last name on billing addresses without it

Fixes #2828 

#### Changes proposed in this Pull Request

* Fix an error when the user do not enter their last name on express checkouts such as Google Pay and Apple Pay, adding a - to their last name during the payment process.









#### Testing instructions


* Add a Stripe test payment method on the Google Chrome settings.

<img width="688" alt="Screen Shot 2022-03-09 at 9 59 44 PM" src="https://user-images.githubusercontent.com/1693223/157569564-49712aea-ee73-43db-a980-657e8be19e34.png">

* After it, the Buy now button should be enabled on WooCommerce stores with Google Pay integration enabled.

<img width="452" alt="Screen Shot 2022-03-09 at 10 02 20 PM" src="https://user-images.githubusercontent.com/1693223/157569659-4c1f69e1-df65-4efa-94e6-1c12a767c3fd.png">

* Click on that button and add a billing address with only the first name.

<img width="556" alt="Screen Shot 2022-03-09 at 10 01 27 PM" src="https://user-images.githubusercontent.com/1693223/157569815-9ad6706b-03b0-4c66-9054-04e17001b552.png">

* Click on the Done button and then the Pay button.

The order should complete as expected, without the changes on this PR it would show an error message asking for the billing address last name as seen in #2828.

One caveat is that this - appear on the user order details, but currently WooCommerce does not support an empty space or an empty last name.

<img width="299" alt="Screen Shot 2022-03-09 at 10 45 44 PM" src="https://user-images.githubusercontent.com/1693223/157570890-850cb418-d0bf-432f-8209-084e4d8a7746.png">




-------------------


**Post merge**
